### PR TITLE
Reconcile Fixes

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Return all metrics validation errors
 - [x] Return artifact parsing errors
 - [x] Return error for invalid file name
 - [x] Use reconcile for refresh
 - [x] Add `visible` to measures and dimensions
 - [x] Rename to existing file returns error

### CODE REVIEW

**Review by begelundmuller — CHANGES_REQUESTED** [review]
Let's not deal with it now, but `RefreshAndReconcile` feels a little misplaced. Maybe because it's such a thin wrapper around `Reconcile`. Or maybe it's just that we haven't really gotten the behavior around source refreshes right (sometimes you want to re-ingest a source no matter what and reconcile if the schema changes, other times you want to fail if the source has changed).

File: runtime/drivers/repo.go [hunk 1]
```diff
@@ -23,3 +24,5 @@ type RepoStore interface {
 type RepoObjectStat struct {
 	LastUpdated time.Time
 }
+
+var FileAlreadyExists = errors.New("file already exists")
```
Comment (begelundmuller) [inline runtime/drivers/repo.go:28]: Go convention is prefixing const errors with `Err`. In this case, consider just using `os.ErrExist`

**Comment by AdityaHegde** [comment]
Lets fix the `visible` vs `skip` in another PR

File: runtime/server/repos_test.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import (
+	"context"
+	"testing"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestServer_Repos(t *testing.T) {
```
Comment (begelundmuller) [inline runtime/server/repos_test.go:11]: There's also `drivers/repo_test.go` for testing things like this

**Review by begelundmuller — APPROVED** [review]
Looks good. Few nits, then can merge

**Pull request merged**